### PR TITLE
Fix random behat on save

### PR DESCRIPTION
**Pull request by pierallard**
Things done in this PR:
- Remove all the "waits" after all the save methods
- Add spin methods waiting for flash message.
- Add a new method for checking flash messages. In fact, the previous one returns true. If I replace it by a functional one, it raises for a few tests. So, I add a new functional one, replace it when tests are not outdated, and depreciate the old one.

### CODE REVIEW

File: features/Context/AssertionContext.php [hunk 1]
```diff
@@ -232,21 +232,41 @@ public function theFieldsShouldBeDisabled($fields)
...
      * @Then /^I should see (?:a )?flash message "([^"]*)"$/
      *
-     * @throws ExpectationException
-     *
      * @return bool
      */
     public function iShouldSeeFlashMessage($text)
     {
         return true;
-//        $this->getMainContext()->wait('$(".flash-messages-holder").length > 0');
```
Comment (juliensnz) [inline features/Context/AssertionContext.php]: finally...

**Comment by jjanvier** [comment]
wahou, really cool PR
gtm

File: features/Context/Page/Base/Form.php [hunk 2]
```diff
@@ -75,14 +72,23 @@ public function saveAndClose()
...
+                $tabs = $this->find('css', $this->elements['Oro tabs']['css']);
+            }
+            if (!$tabs) {
+                $tabs = $this->find('css', $this->elements['Form tabs']['css']);
+            }
+
+            return $tabs;
+        }, 'Could not find any tabs container element');
+
+        $tab_dom = $this->spin(function () use ($tabs, $tab) {
```
Comment (juliensnz) [inline features/Context/Page/Base/Form.php]: $tabDom

**Comment by juliensnz** [comment]
GTM, really nice PR

**Pull request merged**